File: java/relational-database-mapping/src/main/java/bowling/Bowling.java
package bowling;

import cricket.Cricket;

public class Bowling extends Cricket {

    private final double bowlingAverage;

    public Bowling(final String name, final String type, final double battingAverage, final double bowlingAverage) {
        super(name, type, battingAverage);
        this.bowlingAverage = bowlingAverage;
    }

    public double getBowlingAverage() {
        return bowlingAverage;
    }
}

File: typescript/relational-database-mapping/src/bowling/bowling.ts
import {Criket} from "../criket/criket.ts";

export class Bowling extends Criket {
  public readonly bowlingAverage: number;

  constructor(name: string, battingAverage: number, bowlingAverage: number, type: string) {
    super(name, battingAverage, type);
    this.bowlingAverage = bowlingAverage;
  }

  getBowlingAverage(): number {
    return this.bowlingAverage;
  }
}
